emmet-builders/emmet/builders/mobility_screening/migration_graph.py

```diff
@@ -0,0 +1,66 @@
+class MigrationGraphBuilder(MapBuilder):
+    def __init__(
+        self,
+        electrodes: MongoStore,
+        tasks: MongoStore,
+        migration_graphs: MongoStore,
+        ltol: float = 0.4,
+        stol: float = 0.6,
+        angle_tol: float = 15,
+        **kwargs
+    ) -> MapBuilder:
+        self.electrodes = electrodes
+        self.tasks = tasks
+        self.ltol = ltol
+        self.stol = stol
+        self.angle_tol = angle_tol
+        self.migration_graphs = migration_graphs
+        self.wi_entries = {}
```

Comment:
I think getting a dictionary of the relevant working_ion entries is general enough that it might be useful in other contexts. Would it be possible to add this as a part of utils here? https://github.com/materialsproject/emmet/blob/main/emmet-builders/emmet/builders/utils.py

I'm thinking the most general case would be taking a list of working_ion strings (e.g. as specified here: https://github.com/materialsproject/emmet/blob/main/emmet-builders/emmet/builders/materials/electrodes.py#L44) and then returning a dictionary with the lowest energy entry for each working_ion string.

Reply:
And then for this builder, in the init statement it can get the list of all working ions strings (from WORKING_IONS so this is only hard coded in one place) and apply the function to get all the working ion entries. This would also allow the user to check what working_ion entries will be used before running the builder.